packages/sonarwhal/src/lib/utils/npm.ts

```diff
@@ -12,21 +12,43 @@ import * as logger from './logging';
 const debug: debug.IDebugger = d(__filename);
 const npmLoadAsync = promisify(npm.load);
 
-
-const packageExists = () => {
+const getPackageContent = (): string => {
     const packagePath: string = path.join(process.cwd(), 'package.json');
+    let packageContent: string;
+
+    try {
+        packageContent = fs.readFileSync(packagePath, 'utf8'); // eslint-disable-line no-sync
+    } catch (err) {
+        // package.json` doesn't exist.
+        return null;
+    }
 
-    return fs.existsSync(packagePath); // eslint-disable-line no-sync
+    return packageContent;
 };
 
 export const installPackages = (packages: Array<string>): boolean => {
-    const global: boolean = !packageExists();
+    /** Content of package.json in the current directory */
+    const packageContent: string = getPackageContent();
```

Comment:
Based on the feedback above I would do something like:

```ts

const packagePath = misc.findPackageRoot(process.cwd());

if(!packagePath || packagePath !== process.cwd()) {
  // Installation is global
}

const content = misc.loadJSONFile(packagePath); // might need to add `package.json`
...
```

Reply:
@molant One question: What if the `packagePath` exists but the parsing of JSON fails? Should we install the package globally or locally?

Reply:
I think that at that moment we should stop the execution. Something weird is going on and we can't recover from it. Maybe print a message saying that and exit?